client/landing/stepper/declarative-flow/internals/steps-repository/components/setup-form/accent-color-control.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+import { Popover } from '@automattic/components';
+import { hasMinContrast, RGB } from '@automattic/onboarding';
+import { ColorPicker } from '@wordpress/components';
+import { Icon } from '@wordpress/icons';
+import { useI18n } from '@wordpress/react-i18n';
+import classNames from 'classnames';
+import { Dispatch, SetStateAction, useState, useRef } from 'react';
+import FormFieldset from 'calypso/components/forms/form-fieldset';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormInput from 'calypso/components/forms/form-text-input';
+import { tip } from 'calypso/signup/icons';
+
+interface AccentColorControlProps {
+	accentColor: AccentColor;
+	setAccentColor: Dispatch< SetStateAction< AccentColor > >;
+}
+
+export type AccentColor = {
+	hex: string;
+	rgb: RGB;
+	default?: boolean;
+};
+
+/**
+ * Generates an inline SVG for the color picker swatch
+ *
+ * @param color the color in HEX
+ * @returns a value for background-image
+ */
+function generateSwatchSVG( color: string | undefined ) {
+	return `url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Ccircle cx='12' cy='12' r='10' stroke='%23000' stroke-opacity='0.2' stroke-width='1' fill='${ encodeURIComponent(
+		color || '#fff'
+	) }'%3E%3C/circle%3E${
+		// render a line when a color isn't selected
+		! color
+			? `%3Cline x1='18' y1='4' x2='7' y2='20' stroke='%23ccc' stroke-width='1'%3E%3C/line%3E`
+			: ''
+	}%3C/svg%3E")`;
+}
+
+const AccentColorControl = ( { accentColor, setAccentColor }: AccentColorControlProps ) => {
+	const { __ } = useI18n();
+	const [ colorPickerOpen, setColorPickerOpen ] = useState< boolean >( false );
```

Comment:
Neat Tip: the boolean type can be inferred here, i.e. no need to add `<boolean>`